pylhe/__init__.py

```diff
@@ -60,11 +129,50 @@ def readLHE(thefile):
                 for p in particles:
                     particle_objs+=[LHEParticle.fromstring(p)]
                 yield LHEEvent(eventinfo,particle_objs)
-    
+
     except ET.ParseError:
         print "WARNING. Parse Error."
         return
-    
+
+def readLHEWithAttributes(thefile):
+  """
+  Iterate through file, similar to readLHE but also set
+  weights and attributes.
+  """
+  try:
+    for event,element in ET.iterparse(thefile,events=['end']):
+      if element.tag == 'event':
+        eventdict = {}
+        data = element.text.split('\n')[1:-1]
+        eventdata,particles = data[0],data[1:]
+        eventdict['eventinfo']=LHEEventInfo.fromstring(eventdata)
+        eventdict['particles'] = []
+        eventdict['weights'] = {}
+        eventdict['attrib'] = element.attrib
+        for p in particles:
+          if not p.strip().startswith("#"):
+            eventdict['particles']+=[LHEParticle.fromstring(p)]
+        for sub in element:
+            if sub.tag=="rwgt":
+                for r in sub:
+                    if r.tag=="wgt":
+                        eventdict["weights"][r.attrib["id"]] = float(r.text.strip())
+        # yield eventdict
+        yield LHEEvent(eventdict["eventinfo"], eventdict["particles"], eventdict["weights"], eventdict["attrib"])
+
+  except ET.ParseError:
+    print "WARNING. Parse Error."
+    return
```

Comment:
we should change this to `log.warning(..)` instead of print

Reply:
cool!